src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -58,35 +105,219 @@ protected bool Active
-        [DebuggerBrowsable(DebuggerBrowsableState.Never)] // TODO: Remove once https://github.com/dotnet/corefx/issues/5868 is addressed.
         public Socket Client
         {
-            get { return ClientCore; }
-            set { ClientCore = value; }
+            get { return _clientSocket; }
+            set { _clientSocket = value; }
         }
 
-        public bool Connected { get { return ConnectedCore; } }
+        public bool Connected => _clientSocket?.Connected ?? false;
 
         public bool ExclusiveAddressUse
         {
-            get { return ExclusiveAddressUseCore; }
-            set { ExclusiveAddressUseCore = value; }
+            get { return _clientSocket?.ExclusiveAddressUse ?? false; }
+            set
+            {
+                if (_clientSocket != null)
+                {
+                    _clientSocket.ExclusiveAddressUse = value;
+                }
+            }
+        }
+
+        // Connects the Client to the specified port on the specified host.
+        public void Connect(string hostname, int port)
+        {
+            if (NetEventSource.IsEnabled) NetEventSource.Enter(this, hostname);
+
+            if (_cleanedUp)
+            {
+                throw new ObjectDisposedException(GetType().FullName);
+            }
+            if (hostname == null)
+            {
+                throw new ArgumentNullException(nameof(hostname));
+            }
+            if (!TcpValidationHelpers.ValidatePortNumber(port))
+            {
+                throw new ArgumentOutOfRangeException(nameof(port));
+            }
+
+            // Check for already connected and throw here. This check
+            // is not required in the other connect methods as they
+            // will throw from WinSock. Here, the situation is more
+            // complex since we have to resolve a hostname so it's
+            // easier to simply block the request up front.
+            if (_active)
+            {
+                throw new SocketException((int)SocketError.IsConnected);
+            }
+
+            // IPv6: We need to process each of the addresses returned from
+            //       DNS when trying to connect. Use of AddressList[0] is
+            //       bad form.
+            IPAddress[] addresses = Dns.GetHostAddresses(hostname);
+            ExceptionDispatchInfo lastex = null;
+            Socket ipv6Socket = null, ipv4Socket = null;
+
+            try
+            {
+                if (_clientSocket == null)
+                {
+                    if (Socket.OSSupportsIPv4)
+                    {
+                        ipv4Socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+                    }
+                    if (Socket.OSSupportsIPv6)
+                    {
+                        ipv6Socket = new Socket(AddressFamily.InterNetworkV6, SocketType.Stream, ProtocolType.Tcp);
+                    }
```

Comment:
Probably.  This is how it was in Windows and in desktop, but I don't currently see any reason why it can't be made lazy.  Will fix.